### PR TITLE
Basic test coverage

### PR DESCRIPTION
**Pull request by epeach**
Extracts test helper to createDanceAPI.

Adds unit tests for group actions in p5

### CODE REVIEW

File: test/unit/cuesTest.js [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
-const createDanceAPI = () => {
-  return new Promise(resolve => {
-    new DanceParty({
-      moveNames: [],
-      playSound: ({callback}) => callback(),
-      onInit: nativeAPI => resolve(nativeAPI),
-    });
-  });
-};
+const helpers = require ('../helpers/createDanceAPI');
```
Comment (joshlory) [inline test/unit/cuesTest.js]: Tiny nit: extra space after `require`.

File: test/unit/groupTest.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+test('changing dance moves for all updates all dancers', async t => {
+  const nativeAPI = await helpers.createDanceAPI();
+  nativeAPI.play({
+    bpm: 120,
+  });
+  nativeAPI.setAnimationSpriteSheet("CAT", 0, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("CAT", 1, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("BEAR", 0, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("BEAR", 1, {}, ()=> {} );
```
Comment (joshlory) [inline test/unit/groupTest.js]: Tiny style nit: 
```suggestion
  nativeAPI.setAnimationSpriteSheet("BEAR", 1, {}, () => {});
```

(A few other places in this PR as well.)

File: test/unit/groupTest.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  const catSprite = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+  const bearSprite = nativeAPI.makeNewDanceSprite("BEAR", null, {x: 200, y: 200});
+
+
+  t.equal(catSprite.current_move, 0);
+  t.equal(bearSprite.current_move, 0);
+  nativeAPI.changeMoveEachLR('all', 1);
+
+  t.equal(catSprite.current_move, 1);
+  t.equal(bearSprite.current_move, 1);
```
Comment (joshlory) [inline test/unit/groupTest.js]: Noticing that we're not consistent about camelCase in the API — wonder if we should update everywhere?  Not blocking.

File: test/unit/groupTest.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  nativeAPI.setAnimationSpriteSheet("CAT", 0, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("CAT", 1, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("BEAR", 0, {}, ()=> {} );
+  nativeAPI.setAnimationSpriteSheet("BEAR", 1, {}, ()=> {} );
+
+  const catSpriteAlpha = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+  const catSpriteBeta = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+  const bearSprite = nativeAPI.makeNewDanceSprite("BEAR", null, {x: 200, y: 200});
+
+
```
Comment (joshlory) [inline test/unit/groupTest.js]: Tiny style nit: extra new line.

File: test/helpers/createDanceAPI.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const DanceParty = require('../../src/p5.dance');
+
+module.exports = {
+  createDanceAPI : () => {
```
Comment (joshlory) [inline test/helpers/createDanceAPI.js]: Nit: extra space after `createDanceAPI`.

File: test/unit/spriteManipulation.js [hunk 6]
```diff
@@ -68,8 +58,31 @@ test('Sprite dance increments by one and loops for next dance', async t => {
...
+    nativeAPI.setAnimationSpriteSheet("CAT", i, {}, ()=> {} );
+  }
+
+  const sprite = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+
+  //Initial value
+  t.equal(sprite.current_move, 0);
+  nativeAPI.changeMoveLR(sprite, 'rand', 1);
+  //Different value
+  t.notEqual(sprite.current_move, 0);
```
Comment (joshlory) [inline test/unit/spriteManipulation.js]: Not always guaranteed? 😄

I think this might fail 1/12 of the time...

Comment (joshlory) [reply test/unit/spriteManipulation.js]: Ah I see, "rand" does guarantee it changes moves.

**Comment by epeach** [comment]
Thanks @joshlory! Never realized how many strange spaces I use! 
space

**Pull request merged**